src/objective-c/GRPCClient/private/GRPCWrappedCall.m

```diff
@@ -132,8 +132,12 @@ - (instancetype) initWithHandler:(void (^)(NSDictionary *))handler {
     grpc_metadata_array_init(&_headers);
     _op.data.recv_initial_metadata = &_headers;
     if (handler) {
+      // Prevent reference cycle with handler
+      __weak typeof(self) weakSelf = self;
       _handler = ^{
-        NSDictionary *metadata = [NSDictionary grpc_dictionaryFromMetadataArray:_headers];
+        __strong typeof(self) strongSelf = weakSelf;
```

Comment:
No need to retain it, as `_handler` is only ever called within `-finish`, and `self`'s reference count is always > 0 when an instance method is being executed (the compiler makes sure of that).

So `weakSelf->_headers` in the next line is safe.

Same for the other two blocks.